tests/experimental/datasources/test_postgres_datasource.py

```diff
@@ -183,19 +167,33 @@ def test_construct_table_asset_directly_with_splitter(create_source):
 
 @pytest.mark.unit
 def test_datasource_gets_batch_list_no_splitter(create_source):
+    fake_batch_spec = {
+        "batch_identifiers": {},
+        "data_asset_name": "my_asset",
+        "table_name": "my_table",
+        "type": "table",
+    }
+
     def validate_batch_spec(spec: SqlAlchemyDatasourceBatchSpec) -> None:
-        assert spec == {
-            "batch_identifiers": {},
-            "data_asset_name": "my_asset",
-            "table_name": "my_table",
-            "type": "table",
-        }
+        assert spec == fake_batch_spec
 
-    with create_source(
-        validate_batch_spec=validate_batch_spec, dialect="postgresql"
-    ) as source:
+    with create_source() as source:
         asset = source.add_table_asset(name="my_asset", table_name="my_table")
-        source.get_batch_list_from_batch_request(asset.get_batch_request())
+        mock_sqlalchemy_execution_engine = sqlachemy_execution_engine_mock_cls(
+            validate_batch_spec=validate_batch_spec,
+            dialect="postgresql",
+        )()
+        (
+            batch_data,
+            batch_markers,
+        ) = mock_sqlalchemy_execution_engine.get_batch_data_and_markers(
+            batch_spec=fake_batch_spec
+        )
+        with mock.patch(
+            "great_expectations.execution_engine.SqlAlchemyExecutionEngine.get_batch_data_and_markers"
+        ) as get_batch_data_and_markers:
```

Comment:
Don't need to worry about this right now while we are discussing the design, but I'd like to see this refactored into a reusable fixture.